src/fsharp/FSharp.Build/Microsoft.FSharp.Targets

```diff
@@ -400,6 +400,63 @@ this file.
 
     </Target>
 
+    <!--
+
+    The old reference assemblies are no longer installed to `Program Files (x86)\Reference Assemblies\Microsoft\FSharp`.
+
+    To avoid breaking legacy projects the hint path for FSharp.Core will be rewritten if and only if the existing hint
+    path is unchanged from the original legacy templates and if a new reference assembly can be found at the new location.
+
+    -->
+    <Target Name="RedirectFSharpCoreReferenceToNewRedistributableLocation" BeforeTargets="ResolveAssemblyReferences">
+      <PropertyGroup>
+        <_OldRootSdkLocation>$(MSBuildProgramFiles32)\Reference Assemblies\Microsoft\FSharp</_OldRootSdkLocation>
+        <_NewRootSdkLocation>$(VsInstallRoot)\MSBuild\15.0\Bin\FSharpSdk</_NewRootSdkLocation>
+
+        <_CoreRelativeSuffix>.NETCore\$(TargetFSharpCoreVersion)\FSharp.Core.dll</_CoreRelativeSuffix>
+        <_FrameworkRelativeSuffix>.NETFramework\v4.0\$(TargetFSharpCoreVersion)\FSharp.Core.dll</_FrameworkRelativeSuffix>
+        <_PortableRelativeSuffix>.NETPortable\$(TargetFSharpCoreVersion)\FSharp.Core.dll</_PortableRelativeSuffix>
+
+        <!-- .NETCore -->
+        <_OldCoreSdkPath>$(_OldRootSdkLocation)\$(_CoreRelativeSuffix)</_OldCoreSdkPath>
+        <_NewCoreSdkPath>$(_NewRootSdkLocation)\$(_CoreRelativeSuffix)</_NewCoreSdkPath>
+
+        <!-- .NETFramework\v4.0 -->
+        <_OldFrameworkSdkPath>$(_OldRootSdkLocation)\$(_FrameworkRelativeSuffix)</_OldFrameworkSdkPath>
+        <_NewFrameworkSdkPath>$(_NewRootSdkLocation)\$(_FrameworkRelativeSuffix)</_NewFrameworkSdkPath>
+
+        <!-- .NETPortable -->
+        <_OldPortableSdkPath>$(_OldRootSdkLocation)\$(_PortableRelativeSuffix)</_OldPortableSdkPath>
+        <_NewPortableSdkPath>$(_NewRootSdkLocation)\$(_PortableRelativeSuffix)</_NewPortableSdkPath>
+      </PropertyGroup>
+
+      <ItemGroup>
+        <!-- Update references to `.NETCore\*\FSharp.Core.dll`. -->
+        <Reference Update="%(Reference.Identity)"
```

Comment:
Don't use `Update=` inside a target . . . it doesn't do the right thing. https://github.com/Microsoft/msbuild/issues/2835

Reply:
Can you elaborate?  I read the attached issue, but wasn't able to follow.  In my testing the `Update=` seemed to do the right thing; the `<HintPath>` for FSharp.Core was appropriately redirected.